### PR TITLE
Live tests run on build drops

### PR DESCRIPTION
**Pull request by MiYanni**
## Description
**NOTE:** You must wait for https://github.com/Azure/azure-powershell/pull/5301 to be merged before merging this PR.

This is for allowing the automated live tests to run against actual build drops. Updated a few storage tests to no longer fail. Removed using Test-NewAzureStorageAccount in live smoke testing. Allowed live automation to run based on a path to the nupkg modules.

This PR corresponds with https://github.com/Azure/azure-powershell/issues/4439.

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by MiYanni** [comment]
@azuresdkci test this please

File: src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/Common.ps1 [hunk 1]
```diff
@@ -36,7 +36,10 @@ Gets the default location for a provider
 #>
 function Get-ProviderLocation($provider)
 {
-	if ([Microsoft.Azure.Test.HttpRecorder.HttpMockServer]::Mode -ne [Microsoft.Azure.Test.HttpRecorder.HttpRecorderMode]::Playback)
+	$assemblies = [AppDomain]::Currentdomain.GetAssemblies() | Select-Object FullName | ForEach-Object { $_.FullName.Substring(0, $_.FullName.IndexOf(',')) }
+    if ($assemblies -notcontains 'Microsoft.Azure.Test.HttpRecorder.HttpMockServer' `
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/Common.ps1]: I am not following this.  These are types - do you mean to get the type list from each of the loaded assemblies?

File: src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/Common.ps1 [hunk 2]
```diff
@@ -74,7 +77,10 @@ Cleans the created resource groups
 #>
 function Clean-ResourceGroup($rgname)
 {
-    if ([Microsoft.Azure.Test.HttpRecorder.HttpMockServer]::Mode -ne [Microsoft.Azure.Test.HttpRecorder.HttpRecorderMode]::Playback) {
+	$assemblies = [AppDomain]::Currentdomain.GetAssemblies() | Select-Object FullName | ForEach-Object { $_.FullName.Substring(0, $_.FullName.IndexOf(',')) }
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/Common.ps1]: Same comment

File: tools/AutomationTestFramework/ManageScripts/GenerateAndUploadTestHelperModule.ps1 [hunk 3]
```diff
@@ -1,47 +0,0 @@
-# ----------------------------------------------------------------------------------
```
Comment (markcowl) [inline tools/AutomationTestFramework/ManageScripts/GenerateAndUploadTestHelperModule.ps1]: looks like you need to pull the latest from preview

File: tools/AutomationTestFramework/Management/PackageGenerator.ps1 [hunk 4]
```diff
@@ -0,0 +1,143 @@
+# ----------------------------------------------------------------------------------
```
Comment (markcowl) [inline tools/AutomationTestFramework/Management/PackageGenerator.ps1]: I assume these were in the last PR

File: src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/Common.ps1 (hunk 1 above)
Comment (MiYanni) [reply src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/Common.ps1]: In many places, people just use a Try-Catch to detect if it is running live or using mocking. Instead, I check to see if the assemblies are available and then check the playback mode. I had to do this or else this call throws an exception. Basically, the same check that was there before except with type safety.

File: src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/Common.ps1 (hunk 2 above)
Comment (MiYanni) [reply src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/Common.ps1]: See above.

File: tools/AutomationTestFramework/ManageScripts/GenerateAndUploadTestHelperModule.ps1 (hunk 3 above)
Comment (MiYanni) [reply tools/AutomationTestFramework/ManageScripts/GenerateAndUploadTestHelperModule.ps1]: Pulled.

File: tools/AutomationTestFramework/Management/PackageGenerator.ps1 (hunk 4 above)
Comment (MiYanni) [reply tools/AutomationTestFramework/Management/PackageGenerator.ps1]: Pulled now.

**Comment by MiYanni** [comment]
@azuresdkci Retest this please

**Comment by markcowl** [comment]
on demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/399/

**Pull request merged**